Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py

```diff
@@ -1601,3 +1602,73 @@ def test_traceback_in_return_error_debug_mode_off(mocker):
     assert "This is a test string" not in str(demisto.results.call_args)
     assert "Traceback" not in str(demisto.results.call_args)
     assert "some text" in str(demisto.results.call_args)
+
+
+# append_context unit test
+CONTEXT_MOCK = {
+    'str_key': 'str_value',
+    'dict_key': {
+        'key1': 'val1',
+        'key2': 'val2'
+    },
+    'list_key_str': ['val1', 'val2'],
+    'list_key_list': ['val1', 'val2'],
+    'list_key_dict': ['val1', 'val2']
+}
+
+UPDATED_CONTEXT = {
+    'str_key': 'str_data,str_value',
+    'dict_key': {
+        'key1': 'val1',
+        'key2': 'val2',
+        'data_key': 'data_val'
+    },
+    'list_key_str': ['val1', 'val2', 'str_data'],
+    'list_key_list': ['val1', 'val2', 'val1', 'val2'],
+    'list_key_dict': ['val1', 'val2', {'data_key': 'data_val'}]
+
+}
+
+DATA_MOCK_STRING = "str_data"
+DATA_MOCK_LIST = ['val1', 'val2']
+DATA_MOCK_DICT = {
+    'data_key': 'data_val'
+}
+
+STR_KEY = "str_key"
+DICT_KEY = "dict_key"
+
+
+APPEND_CONTEXT_INPUT = [
+    (CONTEXT_MOCK, DATA_MOCK_STRING, STR_KEY, "key = {}, val = {}".format(STR_KEY, UPDATED_CONTEXT[STR_KEY])),
+    (CONTEXT_MOCK, DATA_MOCK_LIST, STR_KEY, "Cannot append data to the existing context"),
+    (CONTEXT_MOCK, DATA_MOCK_DICT, STR_KEY, "Cannot append data to the existing context"),
+
+    (CONTEXT_MOCK, DATA_MOCK_STRING, DICT_KEY, "Cannot append data to the existing context"),
+    (CONTEXT_MOCK, DATA_MOCK_LIST, DICT_KEY, "Cannot append data to the existing context"),
+    (CONTEXT_MOCK, DATA_MOCK_DICT, DICT_KEY, "key = {}, val = {}".format(DICT_KEY, UPDATED_CONTEXT[DICT_KEY])),
+
+    (CONTEXT_MOCK, DATA_MOCK_STRING, 'list_key_str', "key = {}, val = {}".format('list_key_str', UPDATED_CONTEXT['list_key_str'])),
+    (CONTEXT_MOCK, DATA_MOCK_LIST, 'list_key_list', "key = {}, val = {}".format('list_key_list', UPDATED_CONTEXT['list_key_list'])),
+    (CONTEXT_MOCK, DATA_MOCK_DICT, 'list_key_dict', "key = {}, val = {}".format('list_key_dict', UPDATED_CONTEXT['list_key_dict'])),
+]
+
+
+def get_set_context(key, val):
+    from CommonServerPython import return_error
+    return_error("key = {}, val = {}".format(key, val))
+
+
+@pytest.mark.parametrize('context_mock, data_mock, key, expected_answer', APPEND_CONTEXT_INPUT)
+def test_appendContext(mocker, context_mock, data_mock, key, expected_answer):
+    from CommonServerPython import demisto
+    mocker.patch.object(demisto, 'get', return_value=context_mock.get(key))
+    mocker.patch.object(demisto, 'setContext', side_effect=get_set_context)
```

Comment:
why not a function that just returns the key and value

Reply:
appendContext method does not return anything so I had to find some other way to validate this one